DKMS Build Fails on CentOS 9 Stream (5.14.0-412.el9.x86_64)

```➜  ~ cat /var/lib/dkms/gasket/0.0.git.252.c41789f9/build/make.log                           
DKMS make.log for gasket-0.0.git.252.c41789f9 for kernel 5.14.0-412.el9.x86_64 (x86_64)
Sun Feb 11 09:01:33 AM EST 2024
make -C "/lib/modules/5.14.0-412.el9.x86_64/build" M="/var/lib/dkms/gasket/0.0.git.252.c41789f9/build" modules
make[1]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
make[1]: Entering directory '/usr/src/kernels/5.14.0-412.el9.x86_64'
  CC [M]  /var/lib/dkms/gasket/0.0.git.252.c41789f9/build/gasket_core.o
/var/lib/dkms/gasket/0.0.git.252.c41789f9/build/gasket_core.c: In function ‘gasket_register_device’:
/var/lib/dkms/gasket/0.0.git.252.c41789f9/build/gasket_core.c:1846:47: error: passing argument 1 of ‘class_create’ from incompatible pointer type [-Werror=incompatible-pointer-types]
 1846 |     internal->class = class_create(driver_desc->module, driver_desc->name);
      |                                    ~~~~~~~~~~~^~~~~~~~
      |                                               |
      |                                               struct module *
In file included from ./include/linux/device.h:31,
                 from ./include/linux/cdev.h:8,
                 from /var/lib/dkms/gasket/0.0.git.252.c41789f9/build/gasket_core.h:11,
                 from /var/lib/dkms/gasket/0.0.git.252.c41789f9/build/gasket_core.c:12:
./include/linux/device/class.h:230:54: note: expected ‘const char *’ but argument is of type ‘struct module *’
  230 | struct class * __must_check class_create(const char *name);
      |                                          ~~~~~~~~~~~~^~~~
/var/lib/dkms/gasket/0.0.git.252.c41789f9/build/gasket_core.c:1846:23: error: too many arguments to function ‘class_create’
 1846 |     internal->class = class_create(driver_desc->module, driver_desc->name);
      |                       ^~~~~~~~~~~~
In file included from ./include/linux/device.h:31,
                 from ./include/linux/cdev.h:8,
                 from /var/lib/dkms/gasket/0.0.git.252.c41789f9/build/gasket_core.h:11,
                 from /var/lib/dkms/gasket/0.0.git.252.c41789f9/build/gasket_core.c:12:
./include/linux/device/class.h:230:29: note: declared here
  230 | struct class * __must_check class_create(const char *name);
      |                             ^~~~~~~~~~~~
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:299: /var/lib/dkms/gasket/0.0.git.252.c41789f9/build/gasket_core.o] Error 1
make[1]: *** [Makefile:1934: /var/lib/dkms/gasket/0.0.git.252.c41789f9/build] Error 2
make[1]: Leaving directory '/usr/src/kernels/5.14.0-412.el9.x86_64'
make: *** [Makefile:15: all] Error 2```

--- a/src/gasket_core.c
+++ b/src/gasket_core.c
@@ -1840,12 +1840,14 @@ int gasket_register_device(const struct gasket_driver_desc *driver_desc)
 	memset(internal->devs, 0, sizeof(struct gasket_dev *) * GASKET_DEV_MAX);
 
     /* Function signature for `class_create()` is changed in kernel >= 6.4.x
-     * to only accept a single argument.
+     * to only accept a single argument. This change was also backported to
+     * RHEL 9.4, whose kernel is nominally versioned 5.14.0.
      * */
-#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 4, 0)
-    internal->class = class_create(driver_desc->module, driver_desc->name);
-#else
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 4, 0) || \
+    (defined RHEL_RELEASE_CODE && RHEL_RELEASE_CODE >= RHEL_RELEASE_VERSION(9, 4))
     internal->class = class_create(driver_desc->name);
+#else
+    internal->class = class_create(driver_desc->module, driver_desc->name);
 #endif
 
 	if (IS_ERR(internal->class)) {

--- a/src/gasket_interrupt.c
+++ b/src/gasket_interrupt.c
@@ -158,7 +158,8 @@ gasket_handle_interrupt(struct gasket_interrupt_data *interrupt_data,
 	read_lock(&interrupt_data->eventfd_ctx_lock);
 	ctx = interrupt_data->eventfd_ctxs[interrupt_index];
         if (ctx)
-                #if LINUX_VERSION_CODE >= KERNEL_VERSION(6,8,0)
+                #if LINUX_VERSION_CODE >= KERNEL_VERSION(6,8,0) || \
+                (defined RHEL_RELEASE_CODE && RHEL_RELEASE_CODE >= RHEL_RELEASE_VERSION(9, 5))
                         eventfd_signal(ctx);
                 #else
                         eventfd_signal(ctx, 1);

